src/effects/specialeffectchainslots.cpp

```diff
@@ -0,0 +1,197 @@
+                                                 const QString& chainSlotGroup,
+                                                 unsigned int iChainNumber,
+                                                 EffectsManager* pEffectsManager)
+        : EffectChainSlot(chainSlotGroup, iChainNumber, pEffectsManager,
+                          SignalProcessingStage::Prefader, false,
+                          chainSlotGroup) {
+    // Set the chain to be fully wet.
+    m_pControlChainMix->set(1.0);
+    sendParameterUpdate();
+
+    // TODO(rryan): remove.
+    const ChannelHandleAndGroup* handleAndGroup = nullptr;
+    for (const ChannelHandleAndGroup& handle_group :
+             m_pEffectsManager->registeredInputChannels()) {
+        if (handle_group.name() == group) {
+            handleAndGroup = &handle_group;
+            break;
+        }
+    }
+    DEBUG_ASSERT(handleAndGroup != nullptr);
+
+    // Register this channel alone with the chain slot.
+    registerInputChannel(*handleAndGroup);
+    enableForInputChannel(*handleAndGroup);
+
+    // Add a single effect slot
+    addEffectSlot(chainSlotGroup);
+    // DlgPrefEq loads the Effect with loadEffectToGroup
+}
+
+
+QuickEffectChainSlot::QuickEffectChainSlot(const QString& group,
+                                           unsigned int iChainNumber,
+                                           EffectsManager* pEffectsManager)
+        : PerGroupEffectChainSlot(group, formatEffectChainSlotGroup(group),
+                                  iChainNumber, pEffectsManager) {
+    configureForGroup(group);
+}
+
+QString QuickEffectChainSlot::formatEffectChainSlotGroup(const QString& group) {
+    return formatEffectChainSlotGroupString(group);
+}
+
+QString QuickEffectChainSlot::formatEffectChainSlotGroupString(const QString& group) {
+    return QString("[QuickEffectRack1_%1]").arg(group);
+}
+
+void QuickEffectChainSlot::configureForGroup(const QString& group) {
+    // Set the parameter default value to 0.5 (neutral).
+    setSuperParameter(0.5);
+    setSuperParameterDefaultValue(0.5);
+}
+
+QString QuickEffectChainSlot::formatEffectSlotGroup(const QString& group) {
+        return QString("[QuickEffectRack1_%1_Effect1]")
+                .arg(group);
+}
+
+QString QuickEffectChainSlot::formatEffectSlotGroupString(const QString& group) {
+        return formatEffectSlotGroup(group);
+}
+
+
+EqualizerEffectChainSlot::EqualizerEffectChainSlot(const QString& group,
+                                                   unsigned int iChainNumber,
+                                                   EffectsManager* pEffectsManager)
+        : PerGroupEffectChainSlot(group, formatEffectChainSlotGroup(group),
+                                  iChainNumber, pEffectsManager) {
+    configureForGroup(group);
+}
+
+QString EqualizerEffectChainSlot::formatEffectChainSlotGroup(const QString& group) {
+    return formatEffectChainSlotGroupString(group);
```

Comment:
This is overcomplicated. Can't you merge `formatEffectChainSlotGroupString` and `formatEffectChainSlotGroup` into one function?

Reply:
The issue I'm facing is that we cannot have static virtual functions. So currently I've kept both. One solution is we can remove the virtual functions and not call any of the formatting functions in the base class.

Reply:
Why do we need static functions?

Reply:
Oh I see, some of the static functions are used by ControlPickerMenu... :/ Hmm... there must be a better way to organize this...

Reply:
I think your idea to not call any of the formatting functions in the base PerGroupEffectChainSlot class would be the best way around this.

Reply:
There are calls to the formatting functions in ControlPickerMenu and DlgPrefEQ without instantiating an EffectChainSlot. The other workaround can be to make temporary EffectChainSlots to use the formatting functions.